src/Sylius/Component/Registry/DependencyInjection/Compiler/AliasedServicePass.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+
+namespace Sylius\Component\Registry\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+use Sylius\Component\Registry\ServiceRegistryInterface;
+
+/**
+ * This compiler pass will add services to a registered ServiceRegistry
+ * definition.
+ *
+ * @author Daniel Leech <daniel@dantleech.com>
+ */
+class AliasedServicePass implements CompilerPassInterface
+{
+    private $serviceRegistryId;
+    private $serviceTag;
+    private $options;
+
+    /**
+     * @param string $serviceRegistryId
+     * @param string $serviceTag
+     * @param array $options
+     */
+    public function __construct(
+        $serviceRegistryId,
+        $serviceTag,
+        array $options = []
```

Comment:
Well, if there was only 1 option, but I don't like `..., $aliasAttribute = 'alias', $context = null`. As it means that if you want to set the context you need to redefine the alias. Also if we were to add another option later it would become very ugly.

But its my personal preference, I don't mind (very) much making them arguments.